src/framework/extensions/internal/scriptengine.cpp

```diff
@@ -0,0 +1,279 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2024 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+#include "scriptengine.h"
+
+#include <QJSValueIterator>
+
+#include "jsmoduleloader.h"
+
+#include "log.h"
+
+using namespace mu;
+using namespace mu::extensions;
+using namespace mu::api;
+
+ScriptEngine::ScriptEngine()
+{
+    m_engine = new QJSEngine();
+    m_api = new ScriptApi(this, m_engine);
+
+    m_engine->globalObject().setProperty("api", newQObject(m_api));
+
+    m_moduleLoader = new JsModuleLoader(m_engine);
+    m_moduleLoader->pushEngine(this);
+    QJSValue loaderObj = newQObject(m_moduleLoader);
+    QJSValue requireFn = loaderObj.property("require");
```

Comment:
This is necessary for loading modules to work (done like in Node.js CommonJS modules) 
But I just read that there is support for the import expression (it didn’t exist before, but that was a long time ago)
So I think I'll remove this...
in this PR or the next